### PR TITLE
gems: Added mime info path for mimemagic

### PR DESCRIPTION
**Pull request by edude03**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 117702` at 0c797246 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages skipped due to time constraints:</summary>
  <ul>
    <li>gitlab</li>
    <li>gitlab-ee</li>
  </ul>
</details>
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>iruby</li>
    <li>redmine</li>
  </ul>
</details>

File: pkgs/development/ruby-modules/gem-config/default.nix [hunk 1]
```diff
@@ -379,6 +379,10 @@ in
     '';
   };
 
+  mimemagic = attrs: {
+    FREEDESKTOP_MIME_TYPES_PATH="${shared-mime-info}";
```
Comment (SuperSandro2000) [inline pkgs/development/ruby-modules/gem-config/default.nix:383]:
```suggestion
    FREEDESKTOP_MIME_TYPES_PATH="${shared-mime-info}/share/mime/packages/freedesktop.org.xml";
```

According to https://github.com/mimemagicrb/mimemagic#dependencies it should point to the xml file but I did not test this.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by edude03** [comment]
Thanks @SuperSandro2000, I double checked this and you're right. Not sure why it builds anyway 🤔 Looking at the code it should check that the path is a file but it doesn't.

**Review by ryantm — APPROVED** [review]
This change allows it to build for me.

**Comment by andreykaipov** [comment]
I added the latest master branch to my channels but now I can't install. I think this PR is the issue.

> error: attribute 'mimemagic' at /nix/store/pqgaklshxjj2ji6wwya38cifw4xqn4fh-master/master/pkgs/development/ruby-modules/gem-config/default.nix:397:3 already defined at /nix/store/pqgaklshxjj2ji6wwya38cifw4xqn4fh-master/master/pkgs/development/ruby-modules/gem-config/default.nix:167:3

**Comment by kylesferrazza** [comment]
> error: attribute 'mimemagic' at /nix/store/pqgaklshxjj2ji6wwya38cifw4xqn4fh-master/master/pkgs/development/ruby-modules/gem-config/default.nix:397:3 already defined at /nix/store/pqgaklshxjj2ji6wwya38cifw4xqn4fh-master/master/pkgs/development/ruby-modules/gem-config/default.nix:167:3

I am seeing this issue as well when trying to `nix search nixpkgs/master ...` after this PR.

**Comment by ajs124** [comment]
I pushed a revert. This only added a duplicate redundant entry, which broke the eval.

**Comment by ajs124** [comment]
The other commit that added this was 9b30cda2f6e301abc6f39e01902330d981b12a80 in #118736

**Comment by ryantm** [comment]
Sorry! @ajs124 thank you for fixing it.

**Pull request merged**